### ISSUE TITLE
Estimated time longer

### ISSUE THREAD

**Issue by BelgianGuyWithALaser** [Annex-Engineering/klipper_estimator#3]
Hi Dalegaard,

I was using this script and noticed that on my print that SS estimated 9 1/2 hours it went down to 8h 26min.
![estimator 1](https://user-images.githubusercontent.com/75617571/149476432-17e2646c-1594-4266-9f81-8d5fafc1a205.png)

After a print I rechecked the time it needed and this was 8hours 3min
![estimator 2](https://user-images.githubusercontent.com/75617571/149476503-f6fa2676-7c74-4ef6-9cb3-232698608cf4.png)

I also saw you mentioned that it should be withing 1min on a +12 hour print or it would be considered a bug.

I've printed this on a Voron V2 (if you need more info to look into this, let me know)

regards.

**Comment by dalegaard** [comment]
Hi!

Can you send a config(use the `dump-config` command) along with the gcode file? If you don't want it public you can send it to me at `dalegaard@gmail.com`. I'll take a look at what's going on! If you can send me a Klippy.log as well that would be very useful!

Thanks!

**Comment by BelgianGuyWithALaser** [comment]
this is the dump config 
![estimator 3](https://user-images.githubusercontent.com/75617571/149521693-47d54afb-f9c8-4588-8f74-18840bc46f64.png)

as for the gcode, I had to zip it to upload it directly to github : 
[Body2.zip](https://github.com/Annex-Engineering/klipper_estimator/files/7870627/Body2.zip)

and for the klippy log : 
[klippy.log](https://github.com/Annex-Engineering/klipper_estimator/files/7870630/klippy.log)

**Comment by dalegaard** [comment]
Hi!

Thanks for the files. Your config doesn't seem to match up with your klipper config. The config has accel and accel_to_decel at 3000, but the Klipper config has them at 5000 and 2500 respectively. If I change those in the config and re-run I get 8h4m13s which is still a bit more than your actual time(so I'm looking into what could cause that).

Can you verify what settings the actual print was done with?

Best regards,
Lasse

**Comment by BelgianGuyWithALaser** [comment]
Hey,

sorry for the late reply. Yeah sorry was doing some experimenting with my acceleration/deacceleration value's. it was printed with the 3k accel , 3k deaccel.

I might have found the culptrit for the extra time, is it possible that the estimator doesn't count in my whole "start_print" macro? (QGL, nozzle heating, nozzle whipe)

Best regards,
Jordy

**Comment by dalegaard** [comment]
Hi!

The estimator counts all macros as 0 seconds, as it can't know how long they will take. Depending on when Klipper/Moonraker starts counting "Print start time" that will affect how off the result is.

Best regards,
Lasse

**Comment by dalegaard** [comment]
@BelgianGuyWithALaser 

I've discovered that the time shown in the print history in Fluidd(not sure if you are using Fluidd or Mainsail) is the `Total` time, not the `Actual` print time. This means that time will include heatup and everything. From a print I just did:
![2022-01-27-134205_182x96_scrot](https://user-images.githubusercontent.com/344673/151361328-32954aa2-0595-42c9-9f14-df2c322f81aa.png)
![2022-01-27-134308_263x77_scrot](https://user-images.githubusercontent.com/344673/151361331-aceb6765-6152-4a64-8730-3b6293759028.png)

Maybe this is simply what happened in your case?

Best regards,
Lasse
